### PR TITLE
Add encode_check...

### PR DESCRIPTION
**Pull request by wolfsage**

### CODE REVIEW

File: lib/Email/MIME.pm [hunk 1]
```diff
@@ -119,7 +119,15 @@ my $NO_ENCODE_RE = qr/
 /ix;
 
 sub new {
-  my $self = shift->SUPER::new(@_);
+  my ($class, $text, $arg, @rest) = @_;
+  $arg ||= {};
+
+  my $encode_check = exists $arg->{encode_check}
+                   ? delete $arg->{encode_check}
+                   : 1; # Encode::FB_CROAK
```
Comment (pali) [inline lib/Email/MIME.pm]: Rather use that `Encode::FB_CROAK` named constant. `1` is magic number, `FB_CROAK` makes code more readable. Same for documentation. People doing copy+paste of examples, so `FB_CROAK` resp. `FB_DEFAULT` would make their code more readable too...

File: lib/Email/MIME.pm [hunk 2]
```diff
@@ -542,6 +556,34 @@ sub content_type_attribute_set {
...
+
+  $email->encode_check;
+  $email->encode_check_set(0);
+  $email->encode_check_set(Encode::FB_DEFAULT);
+
+Gets/sets the current C<encode_check> setting (default: I<FB_CROAK>).
+This is the parameter passed to L<Encode/"decode"> an L<Encode/"encode">
+when C<body_str()>, C<body_str_set()>, and C<create()> are called.
+
+See L<Encode> for more information.
```
Comment (pali) [inline lib/Email/MIME.pm]: Better link would be directly to Handling Malformed Data section...
I hope this `L<Encode|Encode/Handling Malformed Data>` could work.

**Comment by rjbs** [comment]
Looks like a good first pass.  Tests to show croakage might be nice, too.

File: lib/Email/MIME.pm (hunk 1 above)
Comment (wolfsage) [inline lib/Email/MIME.pm]: Email::MIME already uses '1' directly and I assume it either has a very good reason or wasn't aware of FB_CROAK at the time.

If it's the first, I don't want to break anything, so I think leaving it as 1 is fine for now

File: lib/Email/MIME.pm (hunk 2 above)
Comment (wolfsage) [inline lib/Email/MIME.pm]: Good point, thanks

File: lib/Email/MIME.pm (hunk 1 above)
Comment (pali) [inline lib/Email/MIME.pm]: I do not know about any reason why to not use `Encode::FB_CROAK` today. Maybe at that time `Encode::FB_CROAK` was not defined or author did not know about `Encode::FB_CROAK`. But Email::MIME already depends on Encode 1.98 which `Encode::FB_CROAK` [supports](https://metacpan.org/pod/release/DANKOGAI/Encode-1.98/Encode.pm#Handling-Malformed-Data).

**Comment by rjbs** [comment]
Thanks, in 1.944.

**Pull request closed without merging**